### DISCUSSION TITLE
Implementing Oauth without Client SDK

### DISCUSSION THREAD

**Discussion by lewinkedrs** [pocketbase/pocketbase#6158] [Q&A] [answered]
Hey, does anyone have a code example where they implemented Oauth in Go instead of using one of the client libraries? I am trying to avoid bringing in the js library as a dependency, everything else has seemed pretty straight forward but I can't figure this one out. Any help would be greatly appreciated. Thanks!

**Comment by tigawanna** [comment]
What have you accomplished so far and what are you using on the client?

**Reply by lewinkedrs** [reply]
Hey here is my repo: https://github.com/lewinkedrs/go-pocketbase-template

On the client I am just using go templates rendered by handlers I added to pocketbase, then I use a little bit of htmx. I am thinking maybe I can just call the web api from a go handler but that seems a little hacky to me.

My hope was to just have a button on a client html page that makes an hx-get request to a go handler and then everything is handled server side. I suppose it's not a huge deal to just do this with the JavaScript sdk but was hoping to see if anyone had integrated it without the sdk.

**Reply by tigawanna** [reply, marked as answer]
You can use the manual redirect method  
- First request goes to the url that you build manually with the correct search params that opens up the external login page 
- After logging in you'll get redirected to your redirect url and pocketbase server handles the rest
